internal/app/secretless/handlers/mysql/protocol/protocol.go

```diff
@@ -130,14 +130,101 @@ func UnpackOkResponse(packet []byte) (*OkResponse, error) {
 // HandshakeV10 represents sever's initial handshake packet
 // See https://mariadb.com/kb/en/mariadb/1-connecting-connecting/#initial-handshake-packet
 type HandshakeV10 struct {
-	ProtocolVersion    byte
-	ServerVersion      string
-	ConnectionID       uint32
-	ServerCapabilities uint32
-	AuthPlugin         string
-	Salt               []byte
+	Header                 []byte
+	ProtocolVersion        byte
+	ServerVersion          string
+	ConnectionID           uint32
+	ServerDefaultCollation uint8
+	StatusFlags            uint16
+	ServerCapabilities     uint32
+	AuthPlugin             string
+	Salt                   []byte
 }
 
+// RemoveClientSSL removes Client SSL Capability from Server Handshake
+func (serverHandshake *HandshakeV10) RemoveClientSSL() {
+	serverHandshake.ServerCapabilities &^= ClientSSL
+}
+
+func (serverHandshake *HandshakeV10) Pack() ([]byte, error) {
+	if len(serverHandshake.Header) != 4 {
+		return nil, errors.New("HandshakeV10#Pack: malformed header")
+	}
+
+	var buf bytes.Buffer
+
+	// Write Header (same as the original), ensure 4 bytes
+	buf.Write(serverHandshake.Header)
+
+	// Write ProtocolVersion
+	buf.WriteByte(serverHandshake.ProtocolVersion)
+
+	// Write ServerVersion
+	if err := WriteNullTerminatedString(&buf, serverHandshake.ServerVersion); err != nil {
+		return nil, err
+	}
+
+	// Write Connection ID
+	connectionIDBuf := make([]byte, 4)
+	binary.LittleEndian.PutUint32(connectionIDBuf, serverHandshake.ConnectionID)
+	buf.Write(connectionIDBuf)
+
+	// Write AuthnPluginDataPart1
```

Comment:
I have a similar comment for most of these comments, which is that I'd like more info inline (just a one sentence, assume no prior knowledge, explanation, with an example if relevant).  Also a link in each case to the relevant docs.